testutil/integration/longwait_dkg_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package integration_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	k1 "github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/cluster"
+	"github.com/obolnetwork/charon/dkg"
+	"github.com/obolnetwork/charon/p2p"
+)
+
+var longwaitDKG = flag.Bool("longwait", false, "Enable long-wait DKG integration test")
+
+// ^ : indicates node is online.
+// - : indicates node is offline.
+// Node 0 starts at t=0min and never goes offline.
+// Subsequent nodes start after an interval of 10min.
+// So, Node 1 starts at t=10min, Node 2 starts at t=20min and so on.
+// Each node except Node 0 stops for 2min in every 10min window before restarting again.
+// After the last node is started, no node goes offline and the DKG completes.
+// +========+============+============+============+============+
+// |  Node  |   0-10m    |   10-20m   |   20-30m   |   30-40m   |
+// +========+============+============+============+============+
+// | Node 0 | ^^^^^^^^^^ | ^^^^^^^^^^ | ^^^^^^^^^^ | ^^^^^^^^^^ |
+// +--------+------------+------------+------------+------------+
+// | Node 1 | ---------- | ^^^--^^^^^ | ^^^^^^--^^ | ^^^^^^^^^^ |
+// +--------+------------+------------+------------+------------+
+// | Node 2 | ---------- | ---------- | ^--^^^^^^^ | ^^^^^^^^^^ |
+// +--------+------------+------------+------------+------------+
+// | Node 3 | ---------- | ---------- | ---------- | ^^^^^^^^^^ |
+// +--------+------------+------------+------------+------------+
+
+func TestLongWaitDKG(t *testing.T) {
+	if !*longwaitDKG {
+		t.Skip("Long wait test is disabled")
+	}
+
+	const (
+		threshold = 3
+		numNodes  = 4
+		numVals   = 10
+		relayURL  = "https://0.relay.obol.tech"
+		// The time period after which a new node is started.
+		window = 10 * time.Second
```

Comment:
This constant is too small for a real-world test.

Parametrize `window` and `nodeDownPeriod`, or provide sensible defaults and a `quick-run` flag which overrides them to 10s and 2s respectively.

Reply:
Agreed. I'll modify the defaults to be in `minutes` instead. Will also the `quick-run` flag as you suggested.

However, note that I'm refactoring this PR as the current version contains a bug. Will re-request review once I'm ready